src/nspawn/nspawn-bind-user.c

```diff
@@ -67,6 +69,31 @@ static int write_and_symlink(
         return 0;
 }
 
+static int write_membership(const char *root, const char *user, const char *group) {
+        int r;
+
+        assert(user);
+        assert(group);
+
+        _cleanup_free_ char *membership = strjoin(user, ":", group, ".membership");
+        if (!membership)
+                return log_oom();
+
+        _cleanup_free_ char *p = path_join(root, "/run/host/userdb/", membership);
+        if (!p)
+                return log_oom();
+
+        _cleanup_close_ int fd = open(p, O_WRONLY|O_CREAT|O_CLOEXEC, 0644);
+        if (fd < 0)
+                return log_error_errno(errno, "Failed to create %s: %m", p);
+
+        r = userns_lchown(p, 0, 0);
```

Comment:
maybe better use userns_chown_at() dirctly here. given you already have an fd anyway